app/models/cart.rb

```diff
@@ -48,28 +48,25 @@ def limit_reached?(item_id)
     count_of(item_id) == Item.find(item_id).inventory
   end
 
-  def has_discount?(item_id)
+  def applicable_discount(item_id)
```

Comment:
method finds highest discount for each item that meets minimum quantity requirements